src/EasyAbp.Abp.SettingUi.Application/SettingUiAppService.cs

```diff
@@ -20,189 +21,216 @@
+			}
+		}
+
+		public virtual async Task ResetSettingValues(List<string> settingNames)
+		{
+			foreach (var name in settingNames)
+			{
+				var setting = _settingDefinitionManager.GetOrNull(name);
+				if (setting == null)
+				{
+					continue;
+				}
+
+				await SetSetting(setting, setting.DefaultValue);
+			}
+		}
+
+		private Task SetSetting(SettingDefinition setting, string value)
+		{
+			if (setting.Providers.Any(p => p == UserSettingValueProvider.ProviderName))
+			{
+				return _settingManager.SetForCurrentUserAsync(setting.Name, value);
+			}
+
+			if (setting.Providers.Any(p => p == GlobalSettingValueProvider.ProviderName))
+			{
+				return _settingManager.SetGlobalAsync(setting.Name, value);
+			}
+
+			//Default
+			return _settingManager.SetForCurrentTenantAsync(setting.Name, value);
+		}
+
+		private IDictionary<string, IDictionary<string, string>> GetMergedSettingProperties()
+		{
+			return _fileProvider
+				.GetDirectoryContents(SettingUiConst.SettingPropertiesFileFolder)
+				.Select(content =>
+					_jsonSerializer.Deserialize<IDictionary<string, IDictionary<string, string>>>(content.ReadAsString()))
+				.SelectMany(dict => dict)
+				.ToDictionary(pair => pair.Key, pair => pair.Value);
+		}
+
+		private List<SettingInfo> SetSettingDefinitionProperties(IDictionary<string, IDictionary<string, string>> settingProperties)
+		{
+			var definedSettingUiPermissions = _permissionDefinitionManager.GetPermissions().Where(p => p.Name.StartsWith(SettingUiPermissions.GroupName));
+			var settingInfos = new List<SettingInfo>();
+			var settingDefinitions = _settingDefinitionManager.GetAll();
+			foreach (var settingDefinition in settingDefinitions)
+			{
+				var si = CreateSettingInfo(settingDefinition);
+				if (settingProperties.ContainsKey(si.Name))
+				{
+					// This Setting is defined in the property file,
+					// set its property values from the dictionary
+					var properties = settingProperties[si.Name];
+					foreach (var kv in properties)
+					{
+						// Do not assign the property if it has already been set by the user.
+						if (!si.Properties.ContainsKey(kv.Key))
+						{
+							si.WithProperty(kv.Key, kv.Value);
+						}
+					}
+				}
+
+				// Default group1: Others
+				if (!si.Properties.ContainsKey(SettingUiConst.Group1))
+				{
+					si.WithProperty(SettingUiConst.Group1, SettingUiConst.DefaultGroup);
+				}
+
+				// Default group2: Others
+				if (!si.Properties.ContainsKey(SettingUiConst.Group2))
+				{
+					si.WithProperty(SettingUiConst.Group2, SettingUiConst.DefaultGroup);
+				}
+
+				// Default type: text
+				if (!si.Properties.ContainsKey(SettingUiConst.Type))
+				{
+					si.WithProperty(SettingUiConst.Type, SettingUiConst.DefaultType);
+				}
+
+				var definedPermission = definedSettingUiPermissions.FirstOrDefault(p => p.Name.EndsWith(si.Name));
+				if (definedPermission != null)
+				{
+					si.Permission = definedPermission.Name;
+				}
+
+				settingInfos.Add(si);
```

Comment:
Can we filter out settings without permission here?
This can simplify the processing of the UI layer.

Reply:
It seems work, but I don't know if I misunderstand your intention.

Do authorization in SettingUiAppService.SetSettingDefinitionProperties:

```csharp
var definedPermission = definedSettingUiPermissions.FirstOrDefault(p => p.Name.EndsWith(si.Name));
if (definedPermission != null)
{
      si.Permission = definedPermission.Name;
      if (await _authorizationService.IsGrantedAsync(si.Permission))
      {
             settingInfos.Add(si);
      }
}
```
Then in SettingUiPageContributor.cs，you can simplify the code:

```csharp
public async Task ConfigureAsync(SettingPageCreationContext context)
{
    var settingUiAppService = context.ServiceProvider.GetService<ISettingUiAppService>();
    foreach (var grp in await settingUiAppService.GroupSettingDefinitions())
    {
        context.Groups.Add(new SettingPageGroup(grp.GroupName, grp.GroupDisplayName, typeof(SettingViewComponent), grp));
    }
}
```

Reply:
Hi @linch90. Let me check this.

Reply:
Hi @linch90. SettingUiPageContributor.cs can be simplify like that. But If we want group based permission management, we should check this in SettingUiAppService.GroupSettingDefinitions.

```csharp
// Group the setting definitions
var groups = new List<SettingGroup>();
foreach (var settingGroup in settingDefinitions
	.GroupBy(sd => sd.Properties[SettingUiConst.Group1].ToString())
	.Select(grp => new SettingGroup
	{
		GroupName = grp.Key,
		GroupDisplayName = _localizer[grp.Key],
		SettingInfos = grp.ToList(),
		Permission = $"{SettingUiPermissions.GroupName}.{grp.Key}"
	}))
{
      var definedSettingUiGroupPermission = definedSettingUiPermissions.FirstOrDefault(p => p.Name == settingGroup.Permission);
      if (definedSettingUiGroupPermission == null || await AuthorizationService.IsGrantedAsync(definedSettingUiGroupPermission.Name))
      {
	      groups.Add(settingGroup);
      }
}

return groups;
```

![image](https://user-images.githubusercontent.com/6229029/103293701-6aabd180-4a01-11eb-8727-3162f7de28ae.png)
![image](https://user-images.githubusercontent.com/6229029/103293717-739ca300-4a01-11eb-8713-4191171fec15.png)

Reply:
Hi, I'll update this PR ASAP.

Reply:
Thank you guys, I will check this soon